src/libs/conduit/conduit_node.cpp

```diff
@@ -466,6 +467,7 @@ Node::set_node(const Node &node)
             Schema *curr_schema = &this->m_schema->add_child(*itr);
             size_t idx = (size_t) this->m_schema->child_index(*itr);
             Node *curr_node = new Node();
+            curr_node->set_allocator(m_allocator_id);
```

Comment:
In the current implementation, setting the allocator MUST be first, since its calls `reset()`. I don't like that there is unintuitive ordering.

Reply:
for internal construction, should we have a helper that sets the allocator w/o reset?